packages/js/src/settings/components/search.js

```diff
@@ -29,96 +16,137 @@ const Search = () => {
 					/>
 				</div>
-				{ query.length > 2 && ! isEmpty( results ) && (
-					<Combobox.Options
-						static={ true }
-						className="yst-max-h-80 yst-scroll-pt-11 yst-scroll-pb-2 yst-space-y-2 yst-overflow-y-auto yst-pb-2"
-					>
-						{ map( results, ( groupedItems, route ) => (
-							<li key={ route }>
-								<h2 className="yst-bg-slate-100 yst-py-2.5 yst-px-4 yst-text-xs yst-font-semibold yst-text-slate-900">{ first( groupedItems ).routeLabel }</h2>
-								<ul className="yst-mt-2 yst-text-sm yst-text-slate-800">
+				{ query.length >= QUERY_MIN_CHARS && ! isEmpty( results ) && (
+					<ul className="yst-max-h-80 yst-scroll-pt-11 yst-scroll-pb-2 yst-overflow-y-auto yst-pb-2">
+						{ map( results, ( groupedItems, index ) => (
+							<li key={ groupedItems?.[ 0 ]?.route || `group-${ index }` }>
+								<Title as="h4" size="3" className="yst-bg-slate-100 yst-py-3 yst-px-4">{ first( groupedItems ).routeLabel }</Title>
+								<ul>
 									{ map( groupedItems, ( item, name ) => (
-										<Combobox.Option
-											key={ name }
-											value={ item }
-											className={ ( { active } ) =>
-												classNames( "yst-cursor-default yst-select-none yst-px-4 yst-py-2", active && "yst-bg-primary-600 yst-text-white" )
-											}
-										>
-											{ item.fieldLabel }
-										</Combobox.Option>
+										<li key={ name }>
+											<Link
+												to={ `${ item.route }#${ item.fieldId }` }
+												onClick={ handleNavigate }
+												className="yst-block yst-no-underline yst-text-sm yst-text-slate-800 yst-select-none yst-py-3 yst-px-4 hover:yst-bg-primary-600 hover:yst-text-white focus:yst-bg-primary-600 focus:yst-text-white"
+											>
+												{ item.fieldLabel }
+											</Link>
+										</li>
 									) ) }
 								</ul>
 							</li>
 						) ) }
-					</Combobox.Options>
+					</ul>
 				) }
-				{ query.length <= QUERY_MIN_CHARS && (
-					<div className="yst-border-t yst-border-slate-100 yst-p-6 yst-space-3 yst-text-center yst-text-sm sm:yst-px-14">
+				{ query.length < QUERY_MIN_CHARS && (
+					<div className="yst-border-t yst-border-slate-100 yst-p-6 yst-py-12 yst-space-3 yst-text-center yst-text-sm">
 						<p className="yst-font-semibold yst-text-slate-900">{ __( "Search", "wordpress-seo" ) }</p>
 						<p className="yst-text-slate-500">{ __( "Please enter a search term that is longer than 3 characters.", "wordpress-seo" ) }</p>
 					</div>
 				) }
-				{ query.length > QUERY_MIN_CHARS && isEmpty( results ) && (
-					<div className="yst-border-t yst-border-slate-100 yst-p-6 yst-space-3 yst-text-center yst-text-sm sm:yst-px-14">
+				{ query.length >= QUERY_MIN_CHARS && isEmpty( results ) && (
+					<div className="yst-border-t yst-border-slate-100 yst-p-6 yst-py-12 yst-space-3 yst-text-center yst-text-sm">
 						<p className="yst-font-semibold yst-text-slate-900">{ __( "No results found", "wordpress-seo" ) }</p>
```

Comment:
FYI, as a side-note: When using translations in JS, you need to be sure to only use them after the translations are loaded in. I.e. don't add a constant like `const translation __( "foo", "bar" );`, because that will result in English only. Instead put it in a function so it gets executed later, after the translations are initialized.

Also I think the translations require the strings to be present in that form, or there are problems with extracting them from the code. I.e. doing `__( foo, namespace);` would most likely cause problems. Could be nice to check though again though.